src/update_engine/kernel_verifier_action.cc

```diff
@@ -24,6 +24,15 @@ void KernelVerifierAction::PerformAction() {
   }
   install_plan_ = GetInputObject();
 
+  // If unset then a new kernel was never installed.
+  if (install_plan_.new_kernel_size == 0 &&
+      install_plan_.new_kernel_hash.empty()) {
```

Comment:
It would be a bug if they mismatched which is partially why I opted to check both, if they mismatch it'll fall through and fail when the empty hash fails to match the computed value and so on. Alternatively I could do if on one and add CHECK() for the other if that's preferable.